### PR TITLE
Website: Add blog post for 16.1.0

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

File: _posts/2024-05-14-16.1.0-release.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## C Data Interface notes
+
+
+## Arrow Flight RPC notes
+
+
+## C++ notes
+
+
+## C# notes
```
Comment (raulcd) [inline _posts/2024-05-14-16.1.0-release.md:45]: @CurtHagenlocher @adamreeve can you help adding the C# notes for the 16.1.0 release?

File: _posts/2024-05-14-16.1.0-release.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## Python notes
+
+
+## R notes
+
+
+For more on what’s in the 16.1.0 R package, see the [R changelog][4].
+
+## Ruby and C GLib notes
```
Comment (raulcd) [inline _posts/2024-05-14-16.1.0-release.md:65]: @kou any note here?

File: _posts/2024-05-14-16.1.0-release.md (hunk 1 above)
Comment (adamreeve) [reply _posts/2024-05-14-16.1.0-release.md:45]: How about:
```suggestion
## C# notes
* Recompute a sliced array's null count on demand when it is unknown ([GH-41136](https://github.com/apache/arrow/issues/41136))
* Support writing sliced arrays in the Arrow IPC format ([GH-40517](https://github.com/apache/arrow/issues/40517), [GH-41225](https://github.com/apache/arrow/issues/41225), [GH-41231](https://github.com/apache/arrow/issues/41231))
* Bug fixes for union array behaviour ([GH-41137](https://github.com/apache/arrow/issues/41137), [41140](https://github.com/apache/arrow/issues/41140))
```

File: _posts/2024-05-14-16.1.0-release.md (hunk 2 above)
Comment (kou) [reply _posts/2024-05-14-16.1.0-release.md:65]: We don't have any changes related to Ruby and C GLib. So we can remove this section.

File: _posts/2024-05-14-16.1.0-release.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+## Java notes
+
+
+## JavaScript notes
+
+
+## Python notes
+
+
+## R notes
```
[the remark this answers is not in the document]
Comment (assignUser) [reply _posts/2024-05-14-16.1.0-release.md:60]: AFAIK we only had minor bug fixes in 16.1.0

File: _posts/2024-05-14-16.1.0-release.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+## Go Notes
+
+
+## Java notes
+
+
+## JavaScript notes
+
+
+## Python notes
```
[the remark this answers is not in the document]
Comment (AlenkaF) [reply _posts/2024-05-14-16.1.0-release.md:71]: I think there is no Python issue in the 16.1.0 release so will leave empty.

File: _posts/2024-05-14-16.1.0-release.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+## C++ notes
+
+
+## C# notes
+
+
+## Go Notes
+
+
+## Java notes
```
[the remark this answers is not in the document]
Comment (danepitkin) [reply _posts/2024-05-14-16.1.0-release.md:65]:
```suggestion
## Java notes
* No new changes were added to this release. The functionality is identical to v16.0.0.
```

File: _posts/2024-05-14-16.1.0-release.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+from [**16 distinct contributors**][2]. See the Install Page to learn how to get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## C Data Interface notes
+
+
+## Arrow Flight RPC notes
```
[the remark this answers is not in the document]
Comment (raulcd) [reply _posts/2024-05-14-16.1.0-release.md]: There are no flight changes for 16.1.0. Removing the section.

File: _posts/2024-05-14-16.1.0-release.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+## C# notes
+
+
+## Go Notes
+
+
+## Java notes
+
+
+## JavaScript notes
```
[the remark this answers is not in the document]
Comment (raulcd) [reply _posts/2024-05-14-16.1.0-release.md:64]:
```suggestion
## JavaScript notes

* Store Timestamps in 64 bits ([GH-40959](https://github.com/apache/arrow/issues/40959))
* Update JS dependencies ([GH-40989](https://github.com/apache/arrow/issues/40989))
* Add at() for array like types ([GH-39131](https://github.com/apache/arrow/issues/39131))
* Refactor imports ([GH-39482](https://github.com/apache/arrow/issues/39482))
```

[the remark this answers is not in the document]
Comment (raulcd) [reply _posts/2024-05-14-16.1.0-release.md:64]: @domoritz do you want to expand those notes?

File: _posts/2024-05-14-16.1.0-release.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+## C# notes
+* Recompute a sliced array's null count on demand when it is unknown ([GH-41136](https://github.com/apache/arrow/issues/41136))
+* Support writing sliced arrays in the Arrow IPC format ([GH-40517](https://github.com/apache/arrow/issues/40517), [GH-41225](https://github.com/apache/arrow/issues/41225), [GH-41231](https://github.com/apache/arrow/issues/41231))
+* Bug fixes for union array behaviour ([GH-41137](https://github.com/apache/arrow/issues/41137), [GH-41140](https://github.com/apache/arrow/issues/41140))
+
+## Go Notes
+
+* Enable support for reading date64 from CSV ([GH-41594](https://github.com/apache/arrow/issues/41594))
+* Update MarshalJSON() for Float32 and Float64 to be able to handle NaN, +Inf and -Inf values ([GH-40563](https://github.com/apache/arrow/issues/40563))
```
[the remark this answers is not in the document]
Comment (zeroshade) [reply _posts/2024-05-14-16.1.0-release.md:62]: Thanks! I was gonna get to it but got bogged down in other things, sorry.

This looks good to me

**Review by raulcd — COMMENTED** [review]
I plan to merge this tomorrow and announce the release.

**Pull request merged**